src/main/java/org/ethereum/beacon/discovery/pipeline/handler/WhoAreYouPacketHandler.java

```diff
@@ -50,6 +50,12 @@ public void handle(final Envelope envelope) {
     if (!HandlerUtil.requireField(Field.PACKET_WHOAREYOU, envelope)) {
       return;
     }
+    if (!HandlerUtil.requireField(Field.MASKING_IV, envelope)) {
```

Comment:
This is an interesting one - we do explicitly check for it later and throw a very explicit exception if it's missing (around line 85).

I'd be tempted to move that check up higher so it happens before we start performing any actual work but keep it throwing an exception rather than just ignoring the packet silently as an unmet requirement.